doc/concepts/configuration/configuration_code.rst

```diff
@@ -0,0 +1,341 @@
+..  _configuration_code:
+
+Configuration in code
+=====================
+
+.. box_cfg_legacy_note_start
+
+.. NOTE::
+
+    Starting with the 3.0 version, the recommended way of configuring Tarantool is using a :ref:`configuration file <configuration_file>`.
+    Configuring Tarantool in code is considered a legacy approach.
+
+.. box_cfg_legacy_note_end
+
+This topic covers specifics of configuring Tarantool in code using the ``box.cfg`` API.
+In this case, a configuration is stored in an :ref:`initialization file <index-init_label>` - a Lua script with the specified configuration options.
+You can find all the available options in the :ref:`Configuration reference <box-cfg-params-ref>`.
+
+
+..  _index-init_label:
+
+Initialization file
+-------------------
+
+If the command to :ref:`start Tarantool <configuration_code_run_instance_tarantool>` includes ``LUA_INITIALIZATION_FILE``, then
+Tarantool begins by invoking the Lua program in the file, which by convention
+may have the name ``script.lua``. The Lua program may get further arguments
+from the command line or may use operating-system functions, such as ``getenv()``.
+The Lua program almost always begins by invoking ``box.cfg()``, if the database
+server will be used or if ports need to be opened. For example, suppose
+``script.lua`` contains the lines
+
+..  _index-init-example:
+
+..  code-block:: lua
+
+    #!/usr/bin/env tarantool
+    box.cfg{
+        listen              = os.getenv("LISTEN_URI"),
+        memtx_memory        = 33554432,
+        pid_file            = "tarantool.pid",
+        wal_max_size        = 2500
+    }
+    print('Starting ', arg[1])
+
+and suppose the environment variable LISTEN_URI contains 3301,
+and suppose the command line is ``~/tarantool/src/tarantool script.lua ARG``.
+Then the screen might look like this:
+
+..  code-block:: console
+
+    $ export LISTEN_URI=3301
+    $ ~/tarantool/src/tarantool script.lua ARG
+    ... main/101/script.lua C> Tarantool 2.8.3-0-g01023dbc2
+    ... main/101/script.lua C> log level 5
+    ... main/101/script.lua I> mapping 33554432 bytes for memtx tuple arena...
+    ... main/101/script.lua I> recovery start
+    ... main/101/script.lua I> recovering from './00000000000000000000.snap'
+    ... main/101/script.lua I> set 'listen' configuration option to "3301"
+    ... main/102/leave_local_hot_standby I> ready to accept requests
+    Starting  ARG
+    ... main C> entering the event loop
+
+If you wish to start an interactive session on the same terminal after
+initialization is complete, you can use :ref:`console.start() <console-start>`.
```

Comment:
I guess it is better to suggest the `-i` option. See https://github.com/tarantool/tarantool/issues/7017 for details.